frame/support/src/traits/hooks.rs

```diff
@@ -80,13 +80,22 @@ pub trait OnIdle<BlockNumber> {
 }
 
 #[impl_for_tuples(30)]
-impl<BlockNumber: Clone> OnIdle<BlockNumber> for Tuple {
+impl<BlockNumber: Copy + AtLeast32BitUnsigned> OnIdle<BlockNumber> for Tuple {
 	fn on_idle(n: BlockNumber, remaining_weight: crate::weights::Weight) -> crate::weights::Weight {
+		let on_idle_functions: &[fn(
+			BlockNumber,
+			crate::weights::Weight,
+		) -> crate::weights::Weight] = &[for_tuples!( #( Tuple::on_idle ),* )];
 		let mut weight = 0;
-		for_tuples!( #(
+		let len = on_idle_functions.len();
+		let start_index = n % (len as u32).into();
+		let start_index = start_index.try_into().unwrap_or_else(|_| {
+			panic!("`start_index % len` always fits into `usize`, because `len` can be in maximum `usize::MAX`; qed");
+		});
```

Comment:
```suggestion
		let start_index = start_index.try_into().expect(
			"`start_index % len` always fits into `usize`, because `len` can be in maximum `usize::MAX`; qed");
		);
```

Just because clippy says somehing, it doesn't need to be right.